WIP: Feature/1297 bit bucket server support

piotr-faba-ardigen opened the pull request.

piotr-faba-ardigen committed:
Add bitbucket Server support

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
working version with separate project and repo

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
fully working case

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

pditommaso committed:
Remove verbos logs on readExitFile check

pditommaso committed:
Throw error cause on InovationTargetException

pditommaso committed:
Update readme

piotr-faba-ardigen committed:
Restore asset manager

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
tests added

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
server and endpoint are the same

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
Remove debug prints

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
clean up

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
little fix

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

kjamrog commented on modules/nextflow/src/main/groovy/nextflow/scm/BitbucketServerRepositoryProvider.groovy:
```diff
@@ -0,0 +1,92 @@
+
+    BitbucketServerRepositoryProvider(String name, ProviderConfig config=null) {
+        def parts = name.split('/') as List<String>
+        if( parts.size()!=2 )
```
Very minor thing: missing whitespaces around `!=`

kjamrog commented on modules/nextflow/src/main/groovy/nextflow/scm/BitbucketServerRepositoryProvider.groovy:
```diff
@@ -0,0 +1,92 @@
+        def repo = parts[-1]
+        def project = parts[0]
+        this.project = project
+        this.repository = repo
```
Could be simplified to:
```
this.project = parts[0]
this.repository = parts[1]
```

kjamrog commented on modules/nextflow/src/main/groovy/nextflow/scm/BitbucketServerRepositoryProvider.groovy:
```diff
@@ -0,0 +1,92 @@
+
+    @Override
+    String getContentUrl( String path ) {
+        // "${config.endpoint}/projects/$project/src/${getMainBranch()}/$path"
```
Comment to remove

kjamrog commented on modules/nextflow/src/main/groovy/nextflow/scm/BitbucketServerRepositoryProvider.groovy:
```diff
@@ -0,0 +1,92 @@
+    }
+
+    private String getMainBranchUrl() {
+        return  "${config.endpoint}/rest/api/1.0/projects/$project/repos/${repository}/branches/default"
```
`$project` -> `${project}` - to be consistent

kjamrog commented on modules/nextflow/src/main/groovy/nextflow/scm/BitbucketServerRepositoryProvider.groovy:
```diff
@@ -0,0 +1,92 @@
+    String getCloneUrl() {
+        Map response = invokeAndParseResponse( getEndpointUrl() )
+
+
```
Too many whitespaces

kjamrog commented on modules/nextflow/src/main/groovy/nextflow/scm/BitbucketServerRepositoryProvider.groovy:
```diff
@@ -0,0 +1,92 @@
+        if( !result )
+            throw new IllegalStateException("Missing clone URL for: $project")
+
+
```
Same here

kjamrog commented on modules/nextflow/src/main/groovy/nextflow/scm/RepositoryProvider.groovy:
```diff
@@ -153,6 +153,9 @@ abstract class RepositoryProvider {
                 }
+            case 404:
+                log.debug "Response status: $code -- ${connection.getErrorStream()?.text}"
+                throw new AbortOperationException("Not found. This uri was attempted for api query ${connection}")
```
Do you want to print the whole `connection` object or just the uri?

piotr-faba-ardigen committed:
Code reviewed; fixed message for 404 error

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
remove comment

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

pditommaso committed:
Add enable option to publishDir

pditommaso committed:
Fix non determinist test failure

pditommaso committed:
Update documentation

pditommaso committed:
Add includeInputs test

piotr-faba-ardigen committed:
Merge branch 'master' into feature/1297_BitBucket-Server-support

pditommaso committed:
Fix type in docs

pditommaso committed:
Fix minor issue when reporting invalid entry error msg

pditommaso committed:
Fix nf-tower compile deps

pditommaso committed:
Add nf-tower init request

pditommaso committed:
Add `of` and `fromList` channel factories

pditommaso committed:
Update changelog [ci skip]

pditommaso committed:
[release 19.09.0-edge] Update timestamp and build numbers [ci skip]

pditommaso committed:
[release 19.09.0-edge] Update timestamp and build numbers (2) [ci skip]

piotr-faba-ardigen committed:
Update the code author

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
Documentation updated

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

piotr-faba-ardigen committed:
Merge branch 'master' into feature/1297_BitBucket-Server-support

pditommaso committed:
Add publish task on failure

colemai committed:
Typo (#1313)

lch14forever committed:
Adding ENDPOINT support for cn regions. (#1323)

* Adding ENDPOINT support for cn regions.

Signed-off-by: Chenhao Li <lichenhao.sg@gmail.com>

* Adding ENDPOINT support for cn regions.

Signed-off-by: Chenhao Li <lichenhao.sg@gmail.com>

pditommaso committed:
Add support for Java 12 #1082

This patch allows the execution using Java 12

Note however the Ignite execution still does not
work and required a prior Java version.

pditommaso committed:
Disable stress.nf test

pditommaso committed:
Add limit on max field lengths sent by Tower

t0rrant committed:
shifterimg workaround for ensuring image pulls (#1327)

Signed-off-by: Manuel Torrinha <manuel.torrinha@tecnico.ulisboa.pt>

pditommaso committed:
Merge branch 'master' of github.com:nextflow-io/nextflow

pditommaso committed:
Fix docs typo

pditommaso committed:
Code cleanup

pditommaso committed:
Make Google Pipelines logging less verbose

pditommaso committed:
Allow multiple mounts for the same K8s claim name

This commit allow multiple mounts for the same claim name
reusing the same target volume.

Credits Ólafur Haukur Flygenring

Merges #1248

pditommaso committed:
Revert manifest submodules attribute to generic type

pditommaso committed:
Fix failing test

pditommaso committed:
Fix Broken container environment when enabling metrics #1289

piotr-faba-ardigen committed:
Address code review comment

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>

Commit:
Use a delete function that doesn't follow symlinks (#1337)

Signed-off-by: Ólafur Haukur Flygenring <olafurh@wuxinextcode.com>

pditommaso committed:
Update GIT readme

pditommaso committed:
Normalise gitmodules field to string

pditommaso committed:
Add Tower to integration tests

pditommaso committed:
Add retry ability to Tower client for 500 class errors

pditommaso committed:
Rename TowerObserver to TowerClient

pditommaso committed:
Add projectName to trace init request

pditommaso committed:
Update cleanup script

pditommaso committed:
Add runName validation

pditommaso committed:
Add access token to build file

pditommaso committed:
Fix runName valdation test

piotr-faba-ardigen committed:
Merge branch 'master' into feature/1297_BitBucket-Server-support

piotr-faba-ardigen committed:
Remove new error type in RepositoryProvider

To be provided in separate PR

Signed-off-by: Piotr Faba <piotr.faba@ardigen.com>
